solidity/contracts/utility/PriceOracle.sol

```diff
@@ -47,15 +39,29 @@ contract PriceOracle is IPriceOracle, Utils {
         tokensToOracles[_tokenB] = _tokenBOracle;
     }
 
-    // ensures that the token is supported by the oracle
-    modifier supportedToken(IERC20Token _token) {
-        _supportedToken(_token);
+    // ensures that the provided addresses are valid
+    modifier vaidAddresses(address _address1, address _address2) {
+        _vaidAddresses(_address1, _address2);
+        _;
+    }
+
+    // error message binary size optimization
+    function _vaidAddresses(address _address1, address _address2) internal pure {
+        _validAddress(_address1);
+        _validAddress(_address2);
+        require(_address1 != _address2, "ERR_SAME_ADDRESSES");

```

Comment:
Either `IDENTICAL_ADDRESSES` or `SAME_ADDRESS` (and why do we even check that inside a function called `_vaidAddresses`?).

Reply:
Fixed

Reply:
I meant to say that the name `_vaidAddresses` was not so suitable.